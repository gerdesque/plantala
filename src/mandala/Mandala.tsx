import React, {useEffect, useContext} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { useCanvas } from './useCanvas';
import { IPlant } from '../plant/Plant';
import { AppContext } from '../app-plantala/Context';
import Share from './Share';

interface IMandalaProps {
  selectedPlants: IPlant[]
  isShareable?: boolean
}

const useStyles = makeStyles(() => ({
  canvas: {
    width: '100%',
    height: '100%'
  }
}));

const Mandala = ({ selectedPlants, isShareable = false }: IMandalaProps) => {
  const classes = useStyles();
  const { setPlants, canvasRef, canvasWidth, canvasHeight } = useCanvas();
  const [, imageMultiplier] = useContext(AppContext);

  useEffect(() => {
    setPlants(selectedPlants);
  }, [selectedPlants, setPlants]);

  return (
    <>
      <canvas 
        ref={canvasRef}
        className={classes.canvas}
        width={canvasWidth * imageMultiplier}
        height={canvasHeight * imageMultiplier}
      />
      {isShareable && <Share canvasRef={canvasRef} />}
    </>
  );
};

export default Mandala;
