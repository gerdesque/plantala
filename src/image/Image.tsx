import React from 'react';
import { Grid } from '@material-ui/core';
import Mandala from '../mandala/Mandala';
import { IPlant } from '../plant/Plant';

interface IImageProps {
  selectedPlants: IPlant[]
}

export default function Image({ selectedPlants }: IImageProps) {
  
  return (
    <Grid container justify="center">
      <Mandala selectedPlants={selectedPlants} isShareable={true} />
    </Grid>
  );
}
