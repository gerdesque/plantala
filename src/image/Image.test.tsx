import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Image from './Image';
import Mandala from '../mandala/Mandala';

describe('Image', () => {

  let shallow: any;
  let wrapper: any;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<Image selectedPlants={[]} />);
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a logo', () => {
    expect(wrapper.find(Mandala)).toHaveLength(1);
  });

});
