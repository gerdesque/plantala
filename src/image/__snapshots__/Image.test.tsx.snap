// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  justify="center"
>
  <Mandala
    isShareable={true}
    selectedPlants={Array []}
  />
</WithStyles(ForwardRef(Grid))>
`;
